20 zamodelowanie danych

MichLuszcz opened the pull request:
Połączenie odpowiednich klas uzywając JPA tak żeby stworzyly się odpowiednie relacje w bazie danych. (WIP)

MichLuszcz committed:
Created the necessary entities without relations between them

MichLuszcz committed:
Added getter and setter annotations to comment, review and Teacher

MichLuszcz committed:
(hopefullly) connected courses and students to reviews

MichLuszcz committed:
Changed CrudRepository to JpaRepository in all Repository interfaces

MichLuszcz committed:
Added @Repository annotation to all repos

mGarbowski commented on src/main/java/pl/edu/pw/pap/review/ReviewRepository.java:
```diff
@@ -0,0 +1,8 @@
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface ReviewRepository extends JpaRepository<Review, Long> {
```
Nie wiem czy to zadziała skoro id dla Review jest typu `CourseReviewKey` a nie `Long`

mGarbowski commented on src/main/java/pl/edu/pw/pap/review/Review.java:
```diff
@@ -0,0 +1,80 @@
+    }
+
+    @Override
+    public boolean equals(Object o) {
```
chyba można anotację lomboka `@EqualsAndHashCode`

mGarbowski commented on src/main/java/pl/edu/pw/pap/comment/Comment.java:
```diff
@@ -0,0 +1,31 @@
+import lombok.Getter;
+import lombok.Setter;
+
+//Maybe it's possible to have one interface for reviews and comments that includes the text, likes and dislikes
```
Może być dobry pomysł, można spróbować tak zrobić potem

mGarbowski commented on src/main/java/pl/edu/pw/pap/course/Course.java:
```diff
@@ -18,6 +22,9 @@ public class Course {
     private Long id;
     private String name;
```
Raczej chcemy więcej pól (język, typ kursu, poziom)

mGarbowski commented on src/main/java/pl/edu/pw/pap/review/Review.java:
```diff
@@ -0,0 +1,80 @@
+
+    @ManyToOne
+    @MapsId("userId")
+    @JoinColumn(name = "user_id") //currently ignoring the unresolved column, possibly caused by type of our database
```
IntelliJ będzie to podkreślał jeśli nie jest połączony do bazy danych, która chodzi i ma takie tabele, trzeba to olać albo wyklikać żeby nie podkreślał

mGarbowski commented on src/main/java/pl/edu/pw/pap/review/Review.java:
```diff
@@ -0,0 +1,80 @@
+import java.util.Objects;
+
+
+class CourseReviewKey implements Serializable {
```
Możliwe że trzeba to będzie wyrzucić do oddzielnego pliku tak żeby obie się dało importować i np. pobierać recenzje z bazy po kluczu

MichLuszcz committed:
Changed fetchtype of course reviews for Courses to Eager because it was breaking something when trying to bootrun

MichLuszcz committed:
Connected teachers and courses

MichLuszcz committed:
Created CommandlineRunner for adding a review, and fixed issues relating to creating one

MichLuszcz committed:
added toString method to Review and added printing the review in the CommandlineRunner

MichLuszcz committed:
fixed Review toString

mGarbowski reviewed (commented):
initial review

mGarbowski committed:
changes to model

add comment-review foreign key relation,
appropriate toString methods to avoid endless recursion,
adding dummy data via CommandlineRunner,
moved ReviewKey to separate file

mGarbowski committed:
Merge remote-tracking branch 'origin/20-zamodelowanie-danych' into 20-zamodelowanie-danych

# Conflicts:
#	src/main/java/pl/edu/pw/pap/review/Review.java

mGarbowski committed:
add example db queries

mGarbowski committed:
Merge branch 'main' into 20-zamodelowanie-danych

mGarbowski reviewed (approved):
Poprawiłem i dodałem pare rzeczy tak jak ustaliliśmy @MichLuszcz
Zobacz co ty o tym sądzisz i jeśli jest ok to możesz mergować
